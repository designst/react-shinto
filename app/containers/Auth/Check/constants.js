import createConstant from 'utils/createConstant';

export const AUTH_CHECK_REQUEST = createConstant(__filename, 'AUTH_CHECK_REQUEST');
export const AUTH_CHECK_SUCCESS = createConstant(__filename, 'AUTH_CHECK_SUCCESS');
export const AUTH_CHECK_FAILURE = createConstant(__filename, 'AUTH_CHECK_FAILURE');
