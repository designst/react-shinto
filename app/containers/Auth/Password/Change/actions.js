import {
  AUTH_PASSWORD_CHANGE_REQUEST,
  AUTH_PASSWORD_CHANGE_SUCCESS,
  AUTH_PASSWORD_CHANGE_FAILURE,
} from './constants';

export const passwordChangeRequest = payload => ({
  type: AUTH_PASSWORD_CHANGE_REQUEST,
  payload,
});

export const passwordChangeSuccess = data => ({
  type: AUTH_PASSWORD_CHANGE_SUCCESS,
  data,
});

export const passwordChangeFailure = error => ({
  type: AUTH_PASSWORD_CHANGE_FAILURE,
  error,
});
