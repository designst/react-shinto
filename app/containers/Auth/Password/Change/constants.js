import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_CHANGE_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_REQUEST',
);
export const AUTH_PASSWORD_CHANGE_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_SUCCESS',
);
export const AUTH_PASSWORD_CHANGE_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_CHANGE_FAILURE',
);
