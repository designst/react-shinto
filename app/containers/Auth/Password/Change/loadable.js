/* @flow */

import loadable from 'loadable-components';

import Loading from 'components/Loading';
import ErrorDisplay from 'components/ErrorDisplay';

export default loadable(() => import(/* webpackChunkName: 'auth/password/change' */ './index'), {
  LoadingComponent: Loading,
  ErrorComponent: ErrorDisplay,
});
