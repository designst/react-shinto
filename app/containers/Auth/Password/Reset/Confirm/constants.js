import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_RESET_CONFIRM_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_REQUEST',
);
export const AUTH_PASSWORD_RESET_CONFIRM_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_SUCCESS',
);
export const AUTH_PASSWORD_RESET_CONFIRM_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_CONFIRM_FAILURE',
);
