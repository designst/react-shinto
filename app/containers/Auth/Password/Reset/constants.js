import createConstant from 'utils/createConstant';

export const AUTH_PASSWORD_RESET_REQUEST = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_REQUEST',
);
export const AUTH_PASSWORD_RESET_SUCCESS = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_SUCCESS',
);
export const AUTH_PASSWORD_RESET_FAILURE = createConstant(
  __filename,
  'AUTH_PASSWORD_RESET_FAILURE',
);
