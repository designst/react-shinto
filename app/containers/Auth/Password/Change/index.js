import { compose } from 'redux';
import { connect } from 'react-redux';

import PasswordChange from 'components/Auth/Password/Change';

import { passwordChangeRequest } from './actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  onSubmit: passwordChangeRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(PasswordChange);
