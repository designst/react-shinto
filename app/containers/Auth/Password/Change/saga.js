import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_CHANGE_REQUEST } from 'containers/Auth/Password/Change/constants';

import {
  passwordChangeSuccess,
  passwordChangeFailure,
} from 'containers/Auth/Password/Change/actions';

const logger = createLogger(__filename);

const authPasswordChangeUrl = process.env.SHINTO_AUTH_PASSWORD_CHANGE_API_ENDPOINT;

export function* authPasswordChange(action) {
  const { oldPassword, newPassword, newPasswordConfirm } = action.payload;

  const apiService = yield getContext('apiService');

  logger('Password Change');

  try {
    const response = yield call(apiService.post, authPasswordChangeUrl, {
      oldPassword,
      newPassword,
      newPasswordConfirm,
    });
    yield put(passwordChangeSuccess(response.data));
  } catch (err) {
    yield put(passwordChangeFailure(err));
  }
}

export default function* authPasswordChangeData() {
  yield takeLatest(AUTH_PASSWORD_CHANGE_REQUEST, authPasswordChange);
}
