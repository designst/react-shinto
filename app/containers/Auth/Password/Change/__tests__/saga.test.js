import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import authPasswordChangeData from 'containers/Auth/Password/Change/saga';
import { AUTH_PASSWORD_CHANGE_REQUEST } from 'containers/Auth/Password/Change/constants';

describe('authPasswordChangeData', () => {
  let authPasswordChangeGenerator;

  beforeEach(() => {
    authPasswordChangeGenerator = authPasswordChangeData();

    const selectDescriptor = authPasswordChangeGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authPasswordChangeSuccess action if it requests the data successfully', () => {
    const oldPassword = 'oldPassword';
    const newPassword = 'newPassword';
    const newPasswordConfirm = 'newPassword';

    return expectSaga(authPasswordChangeData)
      .provide({
        getContext: () => ({
          post: url =>
            url === '/auth/password/change'
              ? {
                  data: {
                    oldPassword,
                    newPassword,
                    newPasswordConfirm,
                  },
                }
              : {},
        }),
      })
      .dispatch({
        type: AUTH_PASSWORD_CHANGE_REQUEST,
        payload: {
          oldPassword,
          newPassword,
          newPasswordConfirm,
        },
      })
      .withReducer(authReducer)
      .run();
  });

  it('should dispatch the authPasswordChangeFailure action if it requests the data unsuccessfully', () => {
    const oldPassword = 'oldPassword';
    const newPassword = 'newPassword';
    const newPasswordConfirm = 'newPassword';

    return expectSaga(authPasswordChangeData)
      .dispatch({
        type: AUTH_PASSWORD_CHANGE_REQUEST,
        payload: {
          oldPassword,
          newPassword,
          newPasswordConfirm,
        },
      })
      .withReducer(authReducer)
      .run();
  });
});
