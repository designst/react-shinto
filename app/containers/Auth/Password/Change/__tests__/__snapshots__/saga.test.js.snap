// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authPasswordChangeData should dispatch the authPasswordChangeFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/password/change/constants/AUTH_PASSWORD_CHANGE_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`authPasswordChangeData should dispatch the authPasswordChangeSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/password/change/constants/AUTH_PASSWORD_CHANGE_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
